### PR TITLE
Add end-to-end test to verify resize on keyboard dismissal

### PR DESCRIPTION
**Pull request by cbracken**
Adds an initial integration test that:
1. captures laid-out widget size
2. focuses a text field to show the keyboard and resize the view
3. unfocuses the text field to dismiss the keyboard and resize the view
4. verify that the final widget size matches the initial size

### CODE REVIEW

File: dev/devicelab/bin/tasks/e2e.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```
Comment (Hixie) [inline dev/devicelab/bin/tasks/e2e.dart]: high-level comment: s/e2e/end-to-end/ throughout (files, directories, etc)

File: dev/e2e/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Flutter end-to-end driver tests
```
Comment (Hixie) [inline dev/e2e/README.md]: add a bit more color here

File: dev/e2e/android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,46 @@
+def localProperties = new Properties()
```
Comment (Hixie) [inline dev/e2e/android/app/build.gradle]: do we really need the android/ and ios/ directories?

File: dev/e2e/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```
Comment (Hixie) [inline dev/e2e/lib/main.dart]: maybe move this up one directory and call it keyboard_resize_test.dart or something?

File: dev/devicelab/manifest.yaml [hunk 5]
```diff
@@ -221,6 +228,13 @@ tasks:
 #    required_agent_capabilities: ["has-ios-device"]
 #    flaky: true    
 
+  e2e_ios:
+    description: >
+      Runs end-to-end Flutter tests on iOS.
+    stage: devicelab_ios
+    required_agent_capabilities: ["has-ios-device"]
+    flaky: true
```
Comment (eseidelGoogle) [inline dev/devicelab/manifest.yaml]: What does this line mean?  Are they flaky?

File: dev/e2e/android/app/build.gradle (hunk 3 above)
Comment (Hixie) [inline dev/e2e/android/app/build.gradle]: never mind, i guess we do

File: dev/e2e/lib/main.dart (hunk 4 above)
Comment (Hixie) [inline dev/e2e/lib/main.dart]: well, don't move it up a directory.

File: dev/devicelab/manifest.yaml (hunk 5 above)
Comment (Hixie) [inline dev/devicelab/manifest.yaml]: @eseidelGoogle See the comment at the top of the file.

Comment (cbracken) [inline dev/devicelab/manifest.yaml]: Yegor's suggestion was to mark new tests flaky until we've seen them run a few rounds, then mark unflaky.

In terms of whether they've been flaky in local runs -- no.

**Comment by Hixie** [comment]
LGTM if you rename these so that we can use this directory for future small driver tests like this. (e.g. "e2e" to "end_to_end", "main" to "keyboard_resize", etc)

File: dev/devicelab/bin/tasks/e2e.dart (hunk 1 above)
Comment (cbracken) [inline dev/devicelab/bin/tasks/e2e.dart]: Moved to `/dev/integration_tests/ui`. Naming changed to `integration_ui`

File: dev/e2e/android/app/build.gradle (hunk 3 above)
Comment (cbracken) [inline dev/e2e/android/app/build.gradle]: Yep :(

File: dev/e2e/lib/main.dart (hunk 4 above)
Comment (cbracken) [inline dev/e2e/lib/main.dart]: Done.

File: dev/e2e/README.md (hunk 2 above)
Comment (cbracken) [inline dev/e2e/README.md]: Done.

File: dev/integration_tests/ui/end_to_end.iml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Hixie) [inline dev/integration_tests/ui/end_to_end.iml]: delete this file?

File: dev/integration_tests/ui/pubspec.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+name: integration_ui
+description: Flutter non-plugin UI integration tests.
+
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_driver:
+    sdk: flutter
+dev_dependencies:
+  test: 0.12.20
```
Comment (Hixie) [inline dev/integration_tests/ui/pubspec.yaml]: you can make this "any" if you add a dependency to flutter_test, that way it's easier for us to keep them in sync.

**Comment by Hixie** [comment]
LGTM

File: dev/integration_tests/ui/end_to_end.iml (hunk 6 above)
Comment (cbracken) [inline dev/integration_tests/ui/end_to_end.iml]: Deleted.

File: dev/integration_tests/ui/pubspec.yaml (hunk 7 above)
Comment (cbracken) [inline dev/integration_tests/ui/pubspec.yaml]: Done.

**Pull request merged**